### PR TITLE
Add isentropic relations

### PR DESCRIPTION
**Pull request by charleskawczynski**
- Adds isentropic relations to moist thermo.

I'm not sure what others we need, but it would be nice to change all instances of these hard-coded equations into function calls.

### CODE REVIEW

File: src/Common/MoistThermodynamics/isentropic.jl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+"""
+struct IsentropicProcess end
+
+"""
+    air_pressure_given_θ(θ::FT, Φ::FT, ::IsentropicProcess)
+
+The air pressure for an isentropic process, where
+
+ - `θ` potential temperature
+ - `Φ` gravitation potential
```
Comment (tapios) [inline src/Common/MoistThermodynamics/isentropic.jl]:
```suggestion
 - `Φ` gravitational potential
```

File: src/Common/MoistThermodynamics/isentropic.jl [hunk 2]
```diff
@@ -0,0 +1,44 @@
+export IsentropicProcess
```
Comment (tapios) [inline src/Common/MoistThermodynamics/isentropic.jl]:
```suggestion
export DryAdiabaticProcess
```
(and similar throughout). I'd make clear that these are all dry isentropic (or adiabatic) processes.

File: src/Common/MoistThermodynamics/isentropic.jl [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+"""
+    air_pressure_given_θ(θ::FT, Φ::FT, ::IsentropicProcess)
+
+The air pressure for an isentropic process, where
+
+ - `θ` potential temperature
+ - `Φ` gravitation potential
+"""
+air_pressure_given_θ(θ::FT, Φ::FT, ::IsentropicProcess) where {FT} =
+  FT(MSLP) * (1 - Φ / (θ * FT(cp_d))) ^ (FT(cp_d) / FT(R_d))
```
Comment (tapios) [inline src/Common/MoistThermodynamics/isentropic.jl]: Consider using moist specific heats, gas constants etc. here and elsewhere?

**Comment by tapios** [comment]
Thanks, @charleskawczynski. This looks useful. We'll need such functions in boundary condition/surface flux implementations. I'd use moist thermodynamic quantities (`R_m` etc.) in the calculations, for somewhat greater accuracy. Otherwise, LGTM.

File: src/Common/MoistThermodynamics/isentropic.jl (hunk 3 above)
Comment (charleskawczynski) [reply src/Common/MoistThermodynamics/isentropic.jl]: Wouldn't using moist specific heats contradict `DryAdiabaticProcess`?

Comment (tapios) [reply src/Common/MoistThermodynamics/isentropic.jl]: By ‘dry’ we usually mean unsaturated and non-condensing (as in an ocean is dry :). If you want to be more specific, you can say ‘unsaturated’.

**Pull request merged**